### DISCUSSION TITLE
How to gracefully restart connection

### DISCUSSION THREAD

**Discussion by noah-hwang** [microsoft/msquic#4093] [Q&A]
When the client calls ConnectionStart ，But connect failed，ClientConnectionCallback call QUIC_CONNECTION_EVENT_SHUTDOWN_COMPLETE event。
Can I quickly restart Connection without closing the connection and re-opening it。

Or the connection can only be reopened with the following call：
ConnectionClose->ConnectionOpen->ConnectionStart

The same question applies to stream restart.
thanks!

**Comment by nibanks** [comment]
We do not support restarts. You must open a new connection and try again. Same for streams.

**Reply by noah-hwang** [reply]
Or is there a way I can configure the connection to automatically reconnect, I can't seem to find it in SetParam 
thanks!

**Reply by nibanks** [reply]
No, there is no support for reconnected, whether via manually or automatic. Sorry.

**Reply by noah-hwang** [reply]
Okay, thank you
